zookeeper-server/src/test/java/org/apache/zookeeper/server/NettyServerCnxnTest.java

```diff
@@ -143,6 +155,51 @@ public void testClientResponseStatsUpdate() throws IOException, InterruptedExcep
         }
     }
 
+    @Test
+    public void testNonMTLSLocalConn() throws IOException, InterruptedException, KeeperException {
+        try (ZooKeeper zk = createClient()) {
+            ServerStats serverStats = serverFactory.getZooKeeperServer().serverStats();
+            //2 for local stat connection and this client
+            assertEquals(2, serverStats.getNonMTLSLocalConnCount());
+            assertEquals(0, serverStats.getNonMTLSRemoteConnCount());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testNonMTLSRemoteConn() throws Exception {
+        Channel channel = mock(Channel.class);
+        ChannelId id = mock(ChannelId.class);
+        ChannelFuture success = mock(ChannelFuture.class);
+        ChannelHandlerContext context = mock(ChannelHandlerContext.class);
+        ChannelPipeline channelPipeline = mock(ChannelPipeline.class);
+
+        when(context.channel()).thenReturn(channel);
+        when(channel.pipeline()).thenReturn(channelPipeline);
+        when(success.channel()).thenReturn(channel);
+        when(channel.closeFuture()).thenReturn(success);
+
+        InetSocketAddress address = new InetSocketAddress(0);
+        when(channel.remoteAddress()).thenReturn(address);
+        when(channel.id()).thenReturn(id);
+        NettyServerCnxnFactory factory = new NettyServerCnxnFactory();
+        LeaderZooKeeperServer zks = mock(LeaderZooKeeperServer.class);
+        factory.setZooKeeperServer(zks);
+        Attribute atr = mock(Attribute.class);
+        Mockito.doReturn(atr).when(channel).attr(
+                Mockito.any()
+        );
+        doNothing().when(atr).set(Mockito.any());
+
+        ServerStats.Provider providerMock = mock(ServerStats.Provider.class);
```

Comment:
The version committed to `master` has an additional

```java
when(zks.isRunning()).thenReturn(true);
```

between the two lines above:

https://github.com/p-malik/zookeeper/blob/b842cd4b463a7c3674963f6f66fc392293c1334c/zookeeper-server/src/test/java/org/apache/zookeeper/server/NettyServerCnxnTest.java#L201-L205

Should the same treatment be applied to this branch?

Reply:
No, the code(NettyServerCxnFactory) is different between this branch and master, that why test case is different